simpeg_drivers/potential_fields/magnetic_scalar/params.py

```diff
@@ -15,352 +15,110 @@
-
-    @bz_channel.setter
-    def bz_channel(self, val):
-        self.setter_validator("bz_channel", val, fun=self._uuid_promoter)
-
-    @property
-    def bz_uncertainty(self):
-        return self._bz_uncertainty
+class MagneticScalarInversionParams(BaseInversionData):
+    """
+    Magnetic scalar inversion parameters.
+
+    :param tmi_channel: Total magnetic intensity channel.
+    :param bx_channel: bx channel.
+    :param by_channel: by channel.
+    :param bz_channel: bz channel.
+    :param bxx_channel: bxx channel.
+    :param bxy_channel: bxy channel.
+    :param bxz_channel: bxz channel.
+    :param byy_channel: byy channel.
+    :param byz_channel: byz channel.
+    :param bzz_channel: bzz channel.
+    :param tmi_uncertainty: Total magnetic intensity uncertainty.
+    :param bx_uncertainty: bx uncertainty.
+    :param by_uncertainty: by uncertainty.
+    :param bz_uncertainty: bz uncertainty.
+    :param bxx_uncertainty: bxx uncertainty.
+    :param bxy_uncertainty: bxy uncertainty.
+    :param bxz_uncertainty: bxz uncertainty.
+    :param byy_uncertainty: byy uncertainty.
+    :param byz_uncertainty: byz uncertainty.
+    :param bzz_uncertainty: bzz uncertainty.
+    :param inducing_field_strength: Inducing field strength.
+    :param inducing_field_inclination: Inducing field inclination.
+    :param inducing_field_declination: Inducing field declination.
+    """
 
-    @bz_uncertainty.setter
-    def bz_uncertainty(self, val):
-        self.setter_validator("bz_uncertainty", val, fun=self._uuid_promoter)
+    name: ClassVar[str] = "Magnetic Scalar Inversion"
+    title: ClassVar[str] = "Magnetic Scalar Inversion"
+    default_ui_json: ClassVar[Path] = (
+        assets_path() / "uijson/magnetic_scalar_inversion.ui.json"
+    )
+
+    inversion_type: str = "magnetic scalar"
+    physical_property: str = "susceptibility"
+
+    tmi_channel: FloatData | None = None
+    bx_channel: FloatData | None = None
+    by_channel: FloatData | None = None
+    bz_channel: FloatData | None = None
+    bxx_channel: FloatData | None = None
+    bxy_channel: FloatData | None = None
+    bxz_channel: FloatData | None = None
+    byy_channel: FloatData | None = None
+    byz_channel: FloatData | None = None
+    bzz_channel: FloatData | None = None
+    tmi_uncertainty: float | FloatData | None = None
+    bx_uncertainty: float | FloatData | None = None
+    by_uncertainty: float | FloatData | None = None
+    bz_uncertainty: float | FloatData | None = None
+    bxx_uncertainty: float | FloatData | None = None
+    bxy_uncertainty: float | FloatData | None = None
+    bxz_uncertainty: float | FloatData | None = None
+    byy_uncertainty: float | FloatData | None = None
+    byz_uncertainty: float | FloatData | None = None
+    bzz_uncertainty: float | FloatData | None = None
+    inducing_field_strength: float | FloatData = 50000.0
+    inducing_field_inclination: float | FloatData = 90.0
+    inducing_field_declination: float | FloatData = 0.0
```

Comment:
```suggestion
    inducing_field_strength: float | FloatData
    inducing_field_inclination: float | FloatData
    inducing_field_declination: float | FloatData
```
This should not be defaulted as it is mandatory.